serveradmin/apps/management/commands/createapp.py

```diff
@@ -0,0 +1,60 @@
+"""Serveradmin
+
+Create application for user
+
+Copyright (c) 2021 InnoGames GmbH
+"""
+from os import environ
+
+from django.contrib.auth.models import User
+from django.core.management import BaseCommand, CommandError
+
+from serveradmin.apps.models import Application
+
+
+class Command(BaseCommand):
+    help = 'Create application for user'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--owner', help='Username of application owner')
+        parser.add_argument('--superuser', action='store_true',
+                            help='Create superuser token')
+        parser.add_argument('--non-interactive', action='store_true', help=(
+            'Gather values from SERVERADMIN_TOKEN_OWNER, SERVERADMIN_TOKEN '
+            'and SERVERADMIN_TOKEN_SUPERUSER env variables. '
+            'Any value for SERVERADMIN_TOKEN_SUPERUSER means yes. '
+            'Empty string or absence means no.'
+        ))
+
+    def handle(self, *args, **options):
+        if options['non_interactive']:
+            owner = environ.get('SERVERADMIN_TOKEN_OWNER')
+            token = environ.get('SERVERADMIN_TOKEN', default='')
+            superuser = bool(
+                environ.get('SERVERADMIN_TOKEN_SUPERUSER', default=False))
+        else:
+            if options['owner']:
+                owner = options['owner']
+            else:
+                owner = input('Username of application owner (must exist!): ')
+            token = input('Token (empty for secure auto generated one): ')
+            superuser = options['superuser']
+
+        user = User.objects.filter(username=owner)
+        if not user.exists():
+            raise CommandError(f'No such user {owner} found!')
+
+        app = Application.objects.filter(
+            owner=user.get(), name=f'default app for {owner}')
+        if app.exists():
+            self.stdout.write(self.style.WARNING(
+                f'Default app for {owner} already exists - skipping.'))
+            return
+
+        app = Application()
+        app.name = f'default app for {owner}'
+        app.owner = user.get()
+        app.superuser = superuser
+        if token != '':
```

Comment:
Why comparison with `''` and not just `if not token`?

Reply:
No particular reason other than this is what I expect when the user does not specify a token. Why is `if not token` better ?

Reply:
`!= ''` will create `''` in memory and then checks the comparison between both memory values. `not token` does a direct falseable check on the `token` memory value.
So, 1. it is faster:
```
>>> setup = "tmp = ''"
>>> code = '''not tmp'''
>>> timeit.timeit(stmt=code, setup=setup)
0.024858662999974968
>>> code = "tmp != ''"
>>> timeit.timeit(stmt=code, setup=setup)
0.03584977599996364
```
2. In general the `not token`, so the falsy check is also more pythonic to use. Took me a while to find the statement but here: https://stackoverflow.com/a/9573283/2625488 (in the comment PEP8 ist getting citated)

Reply:
Okay then. I'll grab the tip from stackoverflow and strip the token then beforehand just to be sure nobody tries to create a token consisting of empty string.

Reply:
Actually, the comment about `strip` strings is the comment after the comment I meant. ^^ But, yes, it might be better to first strip the string to not allow tokens consisting just of whitespace.